src/model/core/fuel.jl

```diff
@@ -110,38 +206,104 @@ function fuel!(EP::Model, inputs::Dict, setup::Dict)
-                
+            for y in intersect(resources_with_fuel(dfGen, fuels[f]), SINGLE_FUEL)))
+    
+    @expression(EP, eFuelConsumption[f in 1:NUM_FUEL, t in 1:T],
+        if !isempty(MULTI_FUELS)
+            eFuelConsumption_multi[f, t] + eFuelConsumption_single[f,t]
+        else
+            eFuelConsumption_single[f,t]
+        end)
+
     @expression(EP, eFuelConsumptionYear[f in 1:NUM_FUEL],
         sum(inputs["omega"][t] * EP[:eFuelConsumption][f, t] for t in 1:T))
 
     
     ### Constraint ###
     ### only apply constraint to generators with fuel type other than None
-    @constraint(EP, FuelCalculation[y in setdiff(HAS_FUEL, THERM_COMMIT), t = 1:T],
+
+    @constraint(EP, cFuelCalculation_single[y in intersect(SINGLE_FUEL, setdiff(HAS_FUEL, THERM_COMMIT)), t = 1:T],
         EP[:vFuel][y, t] - EP[:vP][y, t] * dfGen[y, :Heat_Rate_MMBTU_per_MWh] == 0)
 
+    if !isempty(MULTI_FUELS)
+        @constraint(EP, cFuelCalculation_multi[y in intersect(MULTI_FUELS, setdiff(HAS_FUEL, THERM_COMMIT)), t = 1:T],
+                sum(EP[:vMulFuels][y, i, t]/inputs["HEAT_RATES"][i][y] for i in 1:inputs["MAX_NUM_FUELS"]) - EP[:vP][y, t] == 0 
+            )
+    end
+
+
     if !isempty(THERM_COMMIT)        
         # Only apply piecewise fuel consumption to thermal generators in THERM_COMMIT_PWFU set
         THERM_COMMIT_PWFU = inputs["THERM_COMMIT_PWFU"]
         # segemnt for piecewise fuel usage
         if !isempty(THERM_COMMIT_PWFU)
-            segs = 1:inputs["PWFU_Num_Segments"]
-            PWFU_data = inputs["PWFU_data"]
-            slope_cols = inputs["slope_cols"]
-            intercept_cols = inputs["intercept_cols"]
-            segment_intercept(y, seg) = PWFU_data[y, intercept_cols[seg]]
-            segment_slope(y, seg) = PWFU_data[y, slope_cols[seg]]
-            # constraint for piecewise fuel consumption
-            @constraint(EP, PiecewiseFuelUsage[y in THERM_COMMIT_PWFU, t = 1:T, seg in segs],
-            EP[:vFuel][y, t] >= (EP[:vP][y, t] *  segment_slope(y, seg) + 
-            EP[:vCOMMIT][y, t] * segment_intercept(y, seg)))
+            if isempty(MULTI_FUELS)
+                segs = 1:inputs["PWFU_Num_Segments"]
+                PWFU_data = inputs["PWFU_data"]
+                slope_cols = inputs["slope_cols"]
+                intercept_cols = inputs["intercept_cols"]
+                segment_intercept(y, seg) = PWFU_data[y, intercept_cols[seg]]
+                segment_slope(y, seg) = PWFU_data[y, slope_cols[seg]]
+                # constraint for piecewise fuel consumption
+                @constraint(EP, PiecewiseFuelUsage[y in THERM_COMMIT_PWFU, t = 1:T, seg in segs],
+                EP[:vFuel][y, t] >= (EP[:vP][y, t] *  segment_slope(y, seg) + 
+                EP[:vCOMMIT][y, t] * segment_intercept(y, seg)))
+            else
+                error("Multi-fuel option is not available when piece-wise heat rates are used. Please remove multi fuels to avoid this error.")
```

Comment:
I might recommend that this compatibility check be done during the loading step, rather than the model creation step. @lbonaldo 

That would also eliminate one level of `if` statement.

Reply:
Yes, I'd also suggest moving this check in `load_generators_data` (for example, inside `process_piecewisefuelusage!` or `load_multi_fuels_data!`)